src/wallet/wallet.cpp

```diff
@@ -2149,21 +2154,31 @@ void CWallet::ResendWalletTransactions(interfaces::Chain::Lock& locked_chain, in
     if (fFirst)
         return;
 
+    // Get the best block time
+    auto locked_chain = chain().lock();
+    Optional<int> tip_height = locked_chain->getHeight();
+    int64_t best_block_time = tip_height ? locked_chain->getBlockTime(*tip_height) : 0;
```

Comment:
On further thoughts on this, it seems brittle to use the time the miner decided to put in the block header as opposed to our local time, which at least can be assumed to be monotonic.

Reply:
I think you're right that we should use the local time of the block received, but only in order to maintain existing behaviour.

Can you explain how using block time would be brittle?

Reply:
It will hurt privacy even for tx that have a sufficient fee to get into the next block (+5 min). They will be rebroadcast regardless because a miner might have set the the block time to at least 5 minutes into the future.

Similarly, if the block time is set into the past, you might not rebroadcast transactions for quite some time, even if a block has been mined without your txs in the meantime.

Reply:
I'm not really following how this is any worse for privacy, but I've reverted to using the local time of the tip being updated.

Reply:
Not consistently worse, but in some race-conditions, one of which is enough to give an attacker useful observations

* Attacker opens a bunch of new connections to you
* You create tx with fee target of next block
* tx is broadcast (attacker sees it)
* less than 5 minutes pass, enough to mine a new block, but not enough to propagate the tx to miners
* A block is mined with a timestamp of 60 min in the future, without your tx
* Attacker sees the new block and opens a bunch of new connections to you
* previous random wait in ResendWalletTransactions times out
* we send the tx again to the attacker (we wouldn't do this for non-wallet txs, nor if we used local `GetTime()` for the block time)